Sources/AsyncDNSResolver/dnssd/DNSResolver_dnssd.swift

```diff
@@ -115,18 +115,25 @@ struct DNSSD {
                 byteCount: MemoryLayout<QueryReplyHandler>.stride,
                 alignment: MemoryLayout<QueryReplyHandler>.alignment
             )
-            // The handler might be called multiple times so don't deallocate inside `callback`
-            defer { handlerPointer.deallocate() }
 
             handlerPointer.initializeMemory(as: QueryReplyHandler.self, repeating: handler, count: 1)
 
+            // The handler might be called multiple times so don't deallocate inside `callback`
+            defer {
+                let pointer = handlerPointer.assumingMemoryBound(to: QueryReplyHandler.self)
+                pointer.deinitialize(count: 1)
```

Comment:
Without this line of code, we have these leaks:

<img width="1382" alt="image" src="https://github.com/apple/swift-async-dns-resolver/assets/12647725/e3c36e06-edbf-4ac7-bc22-316e075b46db">

The above is modifying `test_queryAAAA` to run `self.resolver.queryAAAA` 100K times.

With this line there were no leaks.